src/selectors/getChildren.ts

```diff
@@ -74,3 +80,38 @@ export const getChildrenRanked = (state: State, context: Context): Child[] =>
 /** Returns the first visible child of a context. */
 export const firstVisibleChild = (state: State, context: Context) =>
   getChildrenSorted(state, context)[0]
+
+/** Checks if a child lies within the cursor path. */
+const isChildInCursor = (state: State, path: Path, simplePath: SimplePath, child: Child | ThoughtContext) => {
+  const showContexts = isContextViewActive(state, pathToContext(path))
+  const childSimplePath = getChildPath(state, child, simplePath, showContexts)
+  const childPath = appendChildPath(state, childSimplePath, path)
+  return state.cursor && equalThoughtRanked(state.cursor[childPath.length - 1], head(childPath))
+}
+
+/** Returns head of context if parent has active context view. */
+const getResolvedValue = (state: State, path: Path, child: Child | ThoughtContext) => {
+  const showContexts = isContextViewActive(state, pathToContext(path))
+  return showContexts
+    ? head((child as ThoughtContext).context)
+    : (child as Child).value
+}
+
+/** Checks if the child is visible and also checks if the child lies within the cursor. */
+const isChildVisibleWithCursorCheck = _.curry((state: State, path: Path, simplePath: SimplePath, child: Child | ThoughtContext) => {
+  return state.showHiddenThoughts ||
+  isChildVisible(state, pathToContext(simplePath), { value: getResolvedValue(state, path, child), rank: child.rank }) ||
+  isChildInCursor(state, path, simplePath, child)
```

Comment:
This is fine, though it would be great to see you push the limits a bit more with FP. Here's a library I made with some simple `and`/`or` predicates that can be used for composition: https://github.com/raineorshine/fp-and-or

Reply:
Thanks! I will take some notes from this.